oidc_provider/settings.py

```diff
@@ -161,12 +168,19 @@ def get(name, import_str=False):
     Helper function to use inside the package.
     """
     value = None
+    default_value = getattr(default_settings, name)
+
     try:
-        value = getattr(default_settings, name)
         value = getattr(settings, name)
     except AttributeError:
-        if value is None and name in default_settings.required_attrs:
+        if name in default_settings.required_attrs:
             raise Exception('You must set ' + name + ' in your settings.')
+    finally:
+        if isinstance(default_value, dict) and value:
```

Comment:
This `finally` block should be moved out from finally block, and be placed as regular block.. 
This is a bad practise and final block shouldn't be used this way (It will be executed even when Exception will be raised)

My proposition:

```
    try:
        value = getattr(settings, name)
    except AttributeError:
        if name in default_settings.required_attrs:
            raise Exception('You must set ' + name + ' in your settings.')

    if isinstance(default_value, dict) and value:
        default_value.update(value)
        value = default_value
    else:
        value = value or default_value
        value = import_from_str(value) if import_str else value
    return value
```